synapse/handlers/sso.py

```diff
@@ -203,8 +213,10 @@ async def handle_redirect_request(
             ap = next(iter(self._identity_providers.values()))
             return await ap.handle_redirect_request(request, client_redirect_url)
 
-        # otherwise, we have a configuration error
-        raise Exception("Multiple SSO identity providers have been configured!")
+        # otherwise, redirect to the IDP picker
+        return "/_synapse/client/pick_idp?" + urlencode(
+            (("redirectUrl", client_redirect_url),)
+        )
```

Comment:
I had a brief thought that we could just return the HTML template here instead of the redirect to avoid encoding the redirectURL, then decoding it, putting it into a template, then submitting it again back to the server...I think this implementation is OK, just seems to involve slightly more back and forth.

Reply:
> I had a brief thought that we could just return the HTML template here

Yeah, we probably could. The main reason I didn't was because `/_matrix/client/r0/login/sso/redirect` is served as a `JsonResource`, so any errors get rendered as json objects, which is a bit silly. Of course, that doesn't mean we handle any errors which do happen during the `/_matrix/client/r0/login/sso/redirect` request any more gracefully :/.